### PR TITLE
Fixed #27914 -- Used qualified name to extract model field path.

### PR DESCRIPTION
**Pull request by ChillarAnand**
Ticket: https://code.djangoproject.com/ticket/27914

### CODE REVIEW

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -191,7 +191,7 @@ def __str__(self):
 
     def __repr__(self):
         """Display the module, class, and name of the field."""
-        path = '%s.%s' % (self.__class__.__module__, self.__class__.__name__)
+        path = '%s.%s' % (self.__class__.__module__, self.__class__.__qualname__)
```
Comment (charettes) [inline django/db/models/fields/__init__.py]: This change isn't tested? I'm not sure it should be done as part of this PR.

**Comment by timgraham** [comment]
@ChillarAnand, do you know what the next steps are here?

**Comment by ChillarAnand** [comment]
@timgraham No. If you or @charettes tell me next steps, I can try.

**Comment by charettes** [comment]
@ChillarAnand sorry about the confusion here. What I meant is that we should audit all `deconstruct()` patterns that use `__class__.__name__` to pass `__class__.__qualname__` instead.

**Comment by ChillarAnand** [comment]
Fixed it. Thanks for the help @charettes @timgraham.

**Comment by timgraham** [comment]
Do those changes need testing? If not, maybe split this into two commits -- one with the change that is tested and one with the other changes?

**Comment by ChillarAnand** [comment]
Splitted into two commits and added a test case.

**Comment by timgraham** [comment]
What I meant is that in the first commit there are more than 10 qualname changes but most can be reverted without any test failures.

**Comment by ChillarAnand** [comment]
My bad. I thought you are [talking about this](https://github.com/django/django/pull/8177#discussion_r105922557). Restructured commits.

**Pull request merged**